### ISSUE TITLE
add header check rule for rogue "strongmail" MTA

### ISSUE THREAD

**Issue by heraklit256** [vstakhov/rspamd#2248]
### Classification (Please choose *one* option):

* [ ] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [ ] Other bug
* [ ] Feature
* [X] Enhancement

### Reproducibility (Please choose *one* option):

* [ ] Always
* [ ] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [X] Not applicable

### Rspamd version:
1.7.4, but applicable to all versions
### Operation system, CPU, memory and environment:
any
### Description (Please provide a descriptive summary of the issue):
There are several spam mails currently not detected by rspamd as such. They are mostly relayed through Gmail and can be easily detected by the following "Received" header:

    Received: from strongmail ([192.168.122.1])
        by mail1.olympiaedge.net (-); Sat, 21 Apr 2018 12:39:23 -0400

Basically, we only need a rule matching for `Received:\s+from\s+strongmail\s+\((.*)\)\n(.*)\(\-\)(.*)`.
However, the Lua regexp string seems to be completely different and rspamd seems to use its own structure of Received headers. Could someone point me into a direction in order to get the corrent regexp string? Thanks.

**Comment by moisseev** [comment]
It could be something like this (untested):
```lua
reconf['STRONGMAIL'] = {
    re = 'Received=/^:from\\s+strongmail\\s+\\(\\[\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\] \\) by \\S+ \\(-\\); /mH'
    score = 6,
    description = 'Sent with "strongmail" MTA',
    group = 'header'
}
```

[Rspamd regexp module](https://rspamd.com/doc/modules/regexp.html)
https://github.com/vstakhov/rspamd/blob/master/rules/regexp/headers.lua

**Comment by vstakhov** [comment]
Using `[[string]]` you can avoid this stupid double escaping there.

**Comment by heraklit256** [comment]
Thanks for the replies. Try to develop a rule for this...

**Comment by heraklit256** [comment]
Does this look sufficient (still unsure about the Regexp part):

    -- Detect rogue MTA "strongmail"
    reconf['STRONGMAIL'] = {
        re = 'Received=/^:from\\s+strongmail\\s+\\(.*\\) by \\s+ \\(-\\).*$/mH',
        score = 6.0,
        description = 'Sent via rogue "strongmail" MTA',
        group = 'header'
    }

What does `/^:from` do instead of `/^from`? Which characters do I need to escape after all?

**Comment by moisseev** [comment]
- `/^:from` is a typo. It should be `/^from`.
- I think `\\(\\[\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\]\\)` should be faster than `\\(.*\\)`
- `\\s` means a "space" character and `\\S` non-space character, so you should use a capital `S` in `by \\S+`
- You can omit `.*$` at the end of regexp.
- You need to escape _special_ characters.
- As @vstakhov suggested, you can avoid double escaping using `[[string]]`.

So the regexp could be like this:
```
re = [[Received=/^from\s+strongmail\s+\(\[\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\]\) by \S+ \(-\); /mH]]
```

To test your rule you can create `rspamd.local.lua`:
```lua
local reconf = config['regexp']
reconf['STRONGMAIL'] = {
    re = [[Received=/^from\s+strongmail\s+\(\[\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\]\) by \S+ \(-\); /mH]],
    score = 6.0,
    description = 'Sent via rogue "strongmail" MTA',
    group = 'header'
}

```

**Comment by heraklit256** [comment]
Thanks for your help. This went into https://github.com/vstakhov/rspamd/pull/2271 .
